src/mam/mod_mam_muc_elasticsearch_arch.erl

```diff
@@ -87,7 +87,7 @@ archive_message(_Result, Host, MessageId, _UserId, RoomJid, FromJID, SourceJid,
         {error, _} = Err ->
             ?ERROR_MSG("event=archive_muc_message_failed server=~s room=~s source=~s mess_id=~p reason=~1000p",
                        [Host, Room, SourceBinJid, MessageId, Err]),
-            ejabberd_hooks:run(mam_drop_message, Host, [Host]),
+            mongoose_hooks:mam_drop_message(Host, ok),
```

Comment:
I understand your confusion. In fact many hooks `mam_*` are used only by metrics and I think we can completely get rid of them. This is subject of a separate PR, though.

On the other hand, not all hooks need accumulator as in `mongoose_acc`. All of them requires some kind of `Acc` because they are run by `run_fold` in the end.